planner/cascades/implementation_rules.go

```diff
@@ -164,3 +167,21 @@ func (r *ImplShow) OnImplement(expr *memo.GroupExpr, reqProp *property.PhysicalP
 	showPhys.SetSchema(logicProp.Schema)
 	return impl.NewShowImpl(showPhys), nil
 }
+
+// ImplSelection is the implementation rule which implements LogicalSelection to PhysicalSelection.
+type ImplSelection struct {
+}
+
+// Match implements ImplementationRule Match interface.
+func (r *ImplSelection) Match(expr *memo.GroupExpr, prop *property.PhysicalProperty) (matched bool) {
+	return true
```

Comment:
the `selection` operator can not guarantee any physical property. We should consider it matched only when `prop.IsEmpty()` is `false`

Reply:
It seems `prop.IsEmpty()` should be `true`?

Reply:
Yeah. Sorry for the misleading.